experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -601,3 +601,92 @@ function createMapProxy<TSchema extends MapSchema>(): SharedTreeMap<TSchema> {
 	);
 	return proxy;
 }
+
+/**
+ * Create a proxy to a {@link SharedTreeObject} that is backed by a raw object node (see {@link createRawObjectNode}).
+ * @param schema - the schema of the object node
+ * @param content - the content to be stored in the raw node.
+ * A copy of content is stored, the input `content` is not modified and can be safely reused in another call to {@link createRawObjectProxy}.
+ * @remarks
+ * Because this proxy is backed by a raw node, it has the same limitations as the node created by {@link createRawObjectNode}.
+ * Most if its properties and methods will error if read/called.
+ */
+export function createRawObjectProxy<TSchema extends ObjectNodeSchema>(
+	schema: TSchema,
+	content: ProxyNode<TSchema, "javaScript">,
+): SharedTreeObject<TSchema> {
+	// Shallow copy the content and then add the type name symbol to it.
+	const contentCopy = { ...content };
+	Object.defineProperty(contentCopy, typeNameSymbol, { value: schema.name });
+	const proxy = createObjectProxy(schema);
+	const editNode = createRawObjectNode(schema, contentCopy);
+	return setEditNode(proxy, editNode);
+}
+
+/**
+ * Given a content tree that is to be inserted into the shared tree, replace all subtrees that were created by factories
+ * (via {@link SharedTreeObjectFactory.create}) with the content that was passed to those factories.
+ * @remarks
+ * This functions works recursively.
+ * Factory-created objects that are nested inside of the content passed to other factory-created objects, and so on, will be in-lined.
+ * This function also adds the hidden {@link typeNameSymbol} of each object schema to the output.
+ * @example
+ * ```ts
+ * const x = foo.create({
+ *   a: 3, b: bar.create({
+ *     c: [baz.create({ d: 5 })]
+ *   })
+ * });
+ * const y = extractFactoryContent(y);
+ * y === {
+ *   [typeNameSymbol]: "foo", a: 3, b: {
+ *     [typeNameSymbol]: "bar", c: [{ [typeNameSymbol]: "baz", d: 5 }]
+ *  }
+ * }
+ * ```
+ */
+export function extractFactoryContent<T extends ProxyNode<TreeNodeSchema, "javaScript">>(
+	content: T,
+): T {
+	if (Array.isArray(content)) {
+		// `content` is an array
+		return content.map(extractFactoryContent) as T;
+	} else if (content instanceof Map) {
+		// `content` is a map
+		const map = new Map();
+		for (const [k, v] of content) {
+			map.set(k, extractFactoryContent(v));
+		}
+		return map as T;
+	} else if (content !== null && typeof content === "object") {
+		const copy: Record<string, unknown> = {};
+		const editNode = tryGetEditNode(content);
```

Comment:
The next few lines from here are the only part of this function that changed - you can't tell from the diff because I had to move the whole function.